src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -549,6 +550,10 @@ public function load(array $configs, ContainerBuilder $container)
         $container->registerForAutoconfiguration(LoggerAwareInterface::class)
             ->addMethodCall('setLogger', [new Reference('logger')]);
 
+        $container->registerAttributeForAutoconfiguration(EventListener::class, static function (ChildDefinition $definition, EventListener $attribute): void {
```

Comment:
Super minor comment. I find this -> `EventListener::class` confusing because it looks like an event listener not like a PHP attribute for an event listener.

A possible solution would be to move it to:

```php
use Symfony\Component\EventDispatcher\EventListenerAttribute;
```

But we probably don't want to do that. I'm not sure if we've discussed the naming of attribute classes. Should we add the `Attribute` suffix to them like we do for controllers, interfaces, etc.?

Reply:
I understand your confusion, however you've picked some "behind the scenes" code for your example. A developer using that feature would usually use that attribute this way:

```php
#[EventListener]
class MyListener
{
    // …
}
```

This is the use-case I'd optimize our attribute naming for: The attribute describes a class that can be used as an event listener. Imagine that same piece of code with a `#[EventListenerAtttribute]`. The `Attribute` suffix feels a bit misplaced or redundant here, imho.

Since attributes are meant to be a replacement for annotations, think about how they are typically named. For instance, this is how you configure a Doctrine entity:

```php
/**
 * @ORM\Entity
 */
class MyEntity
{
    // …
}
```

Again, it's not an `@ORM\EntityAnnotation` and I personally like it that way.

Regarding that piece of code in `FrameworkExtension`: We could probably use a clarifying alias name in code where an attribute class is not used as an attribute, but referred to for background logic.

```php
use Symfony\Component\EventDispatcher\Attribute\EventListener as EventListenerAttribute;
```

Since `FrameworkExtension` will probably see more configuration code like this in the near future and that file has a lot of imports, a convention like this could make the file more readable.

Reply:
I agree with you 👍 

And my confusion is probably because attributes are still so new, so we're good!